src/lib/cache.coffee

```diff
@@ -0,0 +1,83 @@
+###
+
+dnschain
+http://dnschain.net
+
+Copyright (c) 2014 okTurtles Foundation
+
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+###
+
+redis = require 'redis'
+
+module.exports = (dnschain) ->
+    # expose these into our namespace
+    for k of dnschain.globals
+        eval "var #{k} = dnschain.globals.#{k};"
+
+    class ResolverCache
+        constructor: (@dnschain) ->
+            @log = gNewLogger 'CACHE'
+            @log.info "Loading Redis Cache..."
+            if gConf.get('redis:blockchain:enabled') or gConf.get('redis:oldDNS:enabled')
+                @log.info "cache enabled"
+                [host,port] = gConf.get('redis:socket').split(':')
+                @cache = if port?
+                    port = parseInt port
+                    redis.createClient(port, host)
+                else
+                    redis.createClient(host)
+                if gConf.get('redis:blockchain:enabled')
+                    @blockchainEnabled = true
+                if gConf.get('redis:oldDNS:enabled')
+                    @oldDNSEnabled = true
+                    @oldDNSTTL = 600
```

Comment:
You're hard coding the default here and seem to be ignoring the one from the config.